allennlp/models/multitask.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+import inspect
+from typing import Dict, List
+
+from overrides import overrides
+import torch
+
+from allennlp.data import Vocabulary
+from allennlp.modules import Backbone
+from allennlp.models.model import Model
+from allennlp.models.heads import Head
+from allennlp.nn import InitializerApplicator
+
+
+
+# TODO:
+# 1. implement get_metrics
+# 2. implement make_output_human_readable
+# 3. implement selective arguments / which heads to run at training vs. eval, etc.
+
+
+def get_forward_arguments(module: torch.nn.Module) -> List[str]:
+    signature = inspect.signature(module.forward)
+    arguments = []
+    for arg in signature.parameters:
+        if arg == "self":
+            continue
+        arguments.append(arg)
+    return arguments
```

Comment:
```suggestion
return [arg for arg in signature.parameters if arg != "self"]
```

Reply:
Yeah, I thought I was going to need to do more checks inside the loop.  Simpler this way with only one check.

Reply:
Ah, I remember.  We'll probably need to check for optional arguments, so that you still call a head even if you don't have the labels for it (e.g., at prediction time).  So this might need to move back into a for loop in the end.

Reply:
Turns out I solved this another way, so this method can stay as a list comprehension.